### DISCUSSION TITLE
TasmotaClient include Tasmota Modules

### DISCUSSION THREAD

**Discussion by Dees7** [arendst/Tasmota#12732] [Q&A] [answered]
Hello.
I'd like use TasmotaClient. And I'd like use the same modules like in Tasmota (DS18x20, BME, Counter).
There are modules xsns_01_counter, xsns_05_ds18x20, xsns_09_bmp in Tasmota. Is it possible to use (include) this modules to client and use it.

Or rebuild Tasmota proper as client?

**Comment by sfromis** [comment, marked as answer]
"Possible" is in this case not a yes/no question, but about how much work you want to do to port drivers from Tasmota to a different environment. In my estimation it would not be straightforward, but a significant amount of rewrite. Probably easier to use generic Arduino drivers available (many of the Tasmota drivers are calling such anyway).

Much easier to just deploy another cheap ESP8266 running Tasmota.

**Reply by Dees7** [reply]
Is there any way to communicate with tasmota with wire (not wifi)? I have device (on tasmota) and I need connecting one more (far from this one and from wifi). I'd like  use the first esp as relay for the second one.

**Reply by sfromis** [reply]
Well, anything you connect to gpio pins is communicating via wires.

Longer distances over wire are very susceptible to interference, signals "drowning" or just attenuation. For half-long distances, one setup could be an ESP-32 board with ethernet, and possibly also PoE for power. You may also be able to get driver circuits for longer wires, trading speed for length of wires.
